src/resource-stream.ts

```diff
@@ -0,0 +1,101 @@
+/*!
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Transform} from 'stream';
+import {ParsedArguments} from './';
+
+interface ResourceEvents<T> {
```

Comment:
What is the reason to have to define this normal event-emitter/stream stuff?

Reply:
Otherwise the types for 'data' will just be `any` and users would have to explicitly set a type instead of being able to infer one.